### DISCUSSION TITLE
Addition of mobile-like Touch controls to Windows devices for osu!standard. [Feature request for computers]

### DISCUSSION THREAD

**Discussion by LevelsOfTheMind** [ppy/osu#17689] [Ideas] [answered]
### **Note: I am aware that virtual touch screen controls have been requested for mobile, and in fact reference said requests here. This is not a mobile feature request, but a feature request for Windows computers. I have looked extensively for this specific feature request and could not find it.**

I'd like to request the feature of planned touch controls that have been discussed in mobile builds, such as in [#4195](https://github.com/ppy/osu/issues/4195) and #4201, to be put on Windows machines.

The logic behind this is that many Windows players have touch screen computers and play with a touch screen method. Certain windows devices also have a pen with them that can hover over the screen and can be used to tap circles directly on the screen. When playing with either method, sometimes the keyboard is inaccessible (this is especially the case with Surface devices), meaning that streams and fast circles are very difficult to hit with just a pen or fingers on the screen alone. The addition of having an option to turn on virtual touch screen controls would eliminate this issue, allowing pen, finger, and perhaps even screen tablet users to click on virtual buttons to better hit streams and fast circles.

Additionally, because some osu! players have a smaller play area than normal (adjusted by using screen scaling), there could be an option to put the virtual mouse buttons somewhere around the play area so that the touch controls do not interfere with the play area.
Here's a mimic of the virtual buttons outside the play area, closely following the design in #4195 mentioned previously. The white border surrounding the hit circles is not visible in normal play and is only shown here for demonstration purposes, and represents the play area.
The pink border represents the area where virtual touch controls may appear, which in this case is below the play area.

![Touch screen controls mimic](https://user-images.githubusercontent.com/103141406/162049924-dd7c3b36-c4d9-4630-93f9-f22d844acdc1.png)

For players who have a smaller play area than normal, perhaps the Touch screen controls area during play can be customized to be put anywhere surrounding the play area border. Such as, to the right of the play area, to the left of the play area, above the play area, and below the play area. Here's a mimic of what this type of setting would look like. (I don't know the font that osu uses, please excuse the bad font haha). In the example I have above, the touch screen controls are below the play area.

![Touch screen controls scaling mimic](https://user-images.githubusercontent.com/103141406/162050677-d4516bb6-b115-4a3d-9034-0aec7b5ff609.png)

Of course, if a player uses the full screen area for gameplay, the virtual buttons could likely be customized to be put in an area where hit circles are not likely to appear, or just in a very small area on the screen.
This request is specifically for Windows machines as the Linux and Mac operating systems do not support touch screen controls. Though, if touch screen tablets can be connected to both machines, this virtual touch screen feature might be useful to those users as well.

**Comment by bdach** [comment, marked as answer]
did you read the entire thread? https://github.com/ppy/osu/issues/4201#issuecomment-759710412

as for positioning, i'd personally rather let users move the virtual controls around in the skin editor than to add textual settings in the settings sidebar.

**Reply by LevelsOfTheMind** [reply]
Ah, I did not. My bad. This is my first time on github, so I was mostly looking at threads and wasn't looking through the comments too extensively.
